vocabularies/UI.xml

```diff
@@ -55,7 +55,7 @@ a collection of business object instances, e.g. as a list or table.</String>
 
       <!-- Semantic Views / Perspectives -->
 
-      <Term Name="HeaderInfo" Type="UI.HeaderInfoType" AppliesTo="EntityType">
+      <Term Name="HeaderInfo" Type="UI.HeaderInfoType" Nullable="false" AppliesTo="EntityType">
```

Comment:
If I want an Employee's header info to be the same as their Manager's header info, I could write:
```jsonc
"Employee": {
  "$Kind": "EntityType",
  "$BaseType": "self.Manager",
  "Manager": {
    "$Kind": "NavigationProperty",
    "$Type": "self.Manager"
  },
  "@UI.HeaderInfo": {"$Path": "Manager/@UI.HeaderInfo"}, ...
}
```
But if the Employee has no Manager, the `$Path` would evaluate to a forbidden null value.

Shouldn't annotations that might be used with dynamic values be nullable?

Reply:
Correct, but does that make sense for HeaderInfo? How would an Object Page without HeaderInfo look like?

Reply:
Terms that legally can have a null value, for example Common.FilterDefaultValue, are now explicitly marked as Nullable.

Same for terms that are explicitly defined to be used with path expressions.

Reply:
> Correct, but does that make sense for HeaderInfo? How would an Object Page with a null HeaderInfo look like?

The client could use a default, for example list all fields alphabetically.

Reply:
@BerSie what will the client "Fiori Elements" do if it encounters `@UI.HeaderInfo: null`?

Reply:
There might be places where this is handled gracefully, but I wouldn't bet on that. Already today the vocabulary states that the HeaderInfo is mandatory for the "main entity types". And it's not only properties , where can take it e.g. alphabetically from a list, it's also the object type name. From a consumption (and UX) perspective I regards the header information as mandatory.